Harbor 2.1.5 failed to upgrade 2.3.0-rc1

Steps to reproduce:
1. Install Harbor 2.1.5
2. Trigger a scan all job
3. Schedule a scan all policy `0 0 0 */5 * *`
4. Showdown Harbor 2.1.5 and install Harbor 2.3.0
5. After install, `docker ps` to check container status, found harbor-core is inhealthy
6. Check core.log
```
Jun  8 09:31:11 192.168.80.1 core[8614]: 2021-06-08T01:31:11Z [INFO] [/common/dao/base.go:70]: Register database completed
Jun  8 09:31:11 192.168.80.1 core[8614]: 2021-06-08T01:31:11Z [INFO] [/common/dao/pgsql.go:127]: Upgrading schema for pgsql ...
Jun  8 09:31:11 192.168.80.1 core[8614]: 2021-06-08T01:31:11Z [ERROR] [/common/dao/pgsql.go:132]: Failed to upgrade schema, error: "Dirty database version 50. Fix and force version."
Jun  8 09:31:11 192.168.80.1 core[8614]: 2021-06-08T01:31:11Z [FATAL] [/core/main.go:186]: failed to migrate: Dirty database version 50. Fix and force version.
```
7. admin_job
```
registry=# select * from admin_job;
 id |    job_name    | job_kind |                 cron_str                 |  status  |         job_uuid         |    creation_time    
|        update_time         | deleted |  revision  | status_code | job_parameters 
----+----------------+----------+------------------------------------------+----------+--------------------------+---------------------
+----------------------------+---------+------------+-------------+----------------
  1 | IMAGE_SCAN_ALL | Periodic | {"type":"Custom","cron":"0 0 0 */5 * *"} | pending  | 1848d37b832784483d1e6aa2 | 2021-06-08 01:21:14 
| 2021-06-08 01:21:14.549369 | f       |          0 |           0 | null
  2 | IMAGE_SCAN_ALL | Generic  | {"type":"Manual","cron":""}              | finished | 10aef88b55d5c0c8dbb37961 | 2021-06-08 01:21:16 
| 2021-06-08 01:21:16.953697 | f       | 1623115276 |           3 | null
```

--- a/make/photon/db/docker-entrypoint.sh
+++ b/make/photon/db/docker-entrypoint.sh
@@ -34,7 +34,7 @@ if [ "$(ls -A $PGDATA)" ]; then
                 #   There seems to be a postmaster servicing the new cluster.
                 #   Please shutdown that postmaster and try again.
                 #   Failure, exiting
-                $PGBINOLD/pg_ctl -D "$PGDATAOLD" -w start
+                $PGBINOLD/pg_ctl -D "$PGDATAOLD" -w -o "-p 5433" start
                 $PGBINOLD/pg_ctl -D "$PGDATAOLD" -m fast -w stop
                 ./$CUR/upgrade.sh --old-bindir $PGBINOLD --old-datadir $PGDATAOLD --new-datadir $PGDATANEW
                 # it needs to clean the $PGDATANEW on upgrade failure

